src/main.rs

```diff
@@ -1,57 +1,51 @@
 use {
-    clap::{crate_description, crate_name, App, Arg, ArgMatches},
+    clap::{command, Arg, ArgGroup},
     log::*,
     std::fs,
+    strum::VariantNames,
     validator_lab::{
         kubernetes::Kubernetes,
-        release::{BuildConfig, DeployMethod},
+        release::{BuildConfig, BuildType, DeployMethod},
         SolanaRoot,
     },
 };
 
-fn parse_matches() -> ArgMatches<'static> {
-    App::new(crate_name!())
-        .about(crate_description!())
+fn parse_matches() -> clap::ArgMatches {
+    command!()
         .arg(
-            Arg::with_name("cluster_namespace")
+            Arg::new("cluster_namespace")
                 .long("namespace")
-                .short("n")
+                .short('n')
                 .takes_value(true)
                 .default_value("default")
                 .help("namespace to deploy test cluster"),
         )
         .arg(
-            Arg::with_name("deploy_method")
-                .long("deploy-method")
-                .takes_value(true)
-                .possible_values(&["local", "tar", "skip"])
-                .default_value("local")
-                .help("Deploy method. tar, local, skip. [default: local]"),
-        )
-        .arg(
-            Arg::with_name("local-path")
+            Arg::new("local_path")
                 .long("local-path")
                 .takes_value(true)
-                .required_if("deploy-method", "local")
-                .conflicts_with_all(&["tar", "skip"])
-                .help("Path to local agave repo. Required for 'local' deploy method."),
+                .conflicts_with("release_channel")
+                .help("Build validator from local Agave repo. Specify path here."),
         )
         .arg(
-            Arg::with_name("skip_build")
-                .long("skip-build")
-                .help("Disable building for building from local repo"),
-        )
-        .arg(
-            Arg::with_name("debug_build")
-                .long("debug-build")
-                .help("Enable debug build"),
+            Arg::new("build_type")
+                .long("build-type")
+                .takes_value(true)
+                .possible_values(BuildType::VARIANTS)
+                .default_value(BuildType::Release.into())
+                .help("Specifies the build type: skip, debug, or release."),
         )
         .arg(
             Arg::with_name("release_channel")
                 .long("release-channel")
                 .takes_value(true)
-                .required_if("deploy_method", "tar") // Require if deploy_method is "tar"
-                .help("release version. e.g. v1.17.2. Required if '--deploy-method tar'"),
+                .conflicts_with("local_path")
+                .help("Pulls specific release version. e.g. v1.17.2"),
+        )
+        .group(
+            ArgGroup::new("required_group")
+                .args(&["local_path", "release_channel"])
+                .required(true),
```

Comment:
Since you have the group, I think you can remove the `conflicts_with` bit actually

Reply:
oh good point. will update in next PR